lms/djangoapps/teams/static/teams/js/views/teams_tab.js

```diff
@@ -28,27 +60,165 @@
+                       ).bootstrap();
+                       this.mainView = this.tabbedView = new ViewWithHeader({
+                           header: new HeaderView({
+                               model: new HeaderModel({
+                                   description: gettext("See all teams in your course, organized by topic. Join a team to collaborate with other learners who are interested in the same topic as you are."),
+                                   title: gettext("Teams")
                                })
-                           }]
+                           }),
+                           main: new TabbedView({
+                               tabs: [{
+                                   title: gettext('My Teams'),
+                                   url: 'teams',
+                                   view: new TempTabView({text: 'This is the new Teams tab.'})
+                               }, {
+                                   title: gettext('Browse'),
+                                   url: 'browse',
+                                   view: new TopicsView({
+                                       collection: this.topicsCollection,
+                                       router: this.router
+                                   })
+                               }],
+                               router: this.router
+                           })
                        });
-                       Backbone.history.start();
                    },
 
                    render: function() {
-                       this.$el.append(this.headerView.$el);
-                       this.headerView.render();
-                       this.$el.append(this.tabbedView.$el);
-                       this.tabbedView.render();
+                       this.mainView.setElement(this.$el).render();
+                       return this;
+                   },
+
+                   /**
+                    * Render the list of teams for the given topic ID.
+                    */
+                   goToTopic: function (topicID) {
+                       // Lazily load the teams-for-topic view in
+                       // order to avoid making an extra AJAX call.
+                       if (this.teamsView && this.teamsView.main.collection.topic_id === topicID) {
+                           this.mainView = this.teamsView;
+                           this.render();
+                       } else {
+                           var topicPromise = this.getTopic(topicID),
+                               self = this;
+                           // The user tried to go to a topic that doesn't exist
+                           topicPromise.done(function (topic) {
+                               var successCb, teamCollection, headerView;
+                               successCb = function (collection) {
+                                   self.mainView = self.teamsView = new ViewWithHeader({
```

Comment:
I agree that it is good to save the AJAX call. I just don't see why there need to be two variables pointing to the same thing. If there do need to be two, then I'd prefer to not have the double assignment in a single line, as that is a very odd idiom. One simplification would be to have the lazy view creation in its own function, and then self.mainView would just be assigned to the value returned by that function. That would also help by reducing the size of this function which is much too large.